File: csharp/armstrong-numbers/ArmstrongNumbers.cs
using static System.Math;

using System.Collections.Generic;
using System.Linq;

public static class ArmstrongNumbers
{
    public static bool IsArmstrongNumber(int number)
    {
        var digits = number.ToDigits().ToArray();
        int sumOfPowers = digits.Sum(e => (int)Pow(e, digits.Length));

        return sumOfPowers == number;
    }
}

public static class IntegerExtentions
{
    public static IEnumerable<int> ToDigits(this int number)
    {
        if (number == 0)
        {
            yield return 0;
        }

        while (number != 0)
        {
            yield return number % 10;
            number /= 10;
        }
    }
}

File: javascript/armstrong-numbers/armstrong-numbers.js
class ArmstrongNumber {
    constructor() { }

    validate (number) {
        if (number < 0) {
            return false;
        }

        let digits = this._toDigitArray(number);
        let exponent = digits.length;

        return digits.reduce((a, b) => a += Math.pow(b, exponent), 0) === number;
    }

    _toDigitArray(number) {
        let digits = [];

        if (number === 0) {
            digits.push(number);
        }
        else {
            while (number !== 0) {
                digits.push(number % 10);
                number = Math.floor(number / 10);
            }
        }

        return digits;
    }
}

module.exports = new ArmstrongNumber();
